citesphere/src/main/java/edu/asu/diging/citesphere/api/v1/user/JobInfoController.java

```diff
@@ -54,24 +57,37 @@ public ResponseEntity<String> getProfile(@RequestHeader HttpHeaders headers) {
         IJob job = jobManager.findJob(tokenContents.getJobId());
         IZoteroToken zoteroToken = tokenManager.getToken(userManager.findByUsername(job.getUsername()));
         
-        ObjectMapper mapper = new ObjectMapper();
-        ObjectNode node = mapper.createObjectNode();
-        node.put("zotero", zoteroToken.getToken());
-        node.put("zoteroId", zoteroToken.getUserId());
-        node.put("username", job.getUsername());
+        Map<String, Object> response = new HashMap<>();
+        response.put("zotero", zoteroToken.getToken());
+        response.put("zoteroId", zoteroToken.getUserId());
+        response.put("username", job.getUsername());
         // FIXME: ugly, needs better solution
         if (job instanceof IUploadJob) {
-            node.put("groupId", ((IUploadJob)job).getCitationGroup());
-        }
-        if (job instanceof IExportJob) {
-            IExportTask exportTask = exportTaskManager.get(((IExportJob)job).getTaskId());
-            node.put("groupId", exportTask.getGroupId());
-            node.put("collectionId", exportTask.getCollectionId());
-            node.put("exportType", exportTask.getExportType().name());
-            node.put("taskId", exportTask.getId());
+            handleUploadJob(response, (IUploadJob) job);
+        } else if (job instanceof IExportJob) {
+            handleExportJob(response, (IExportJob) job);
+        } else if (job instanceof IImportCrossrefJob) {
+            handleImportCrossrefJob(response, (IImportCrossrefJob) job);
```

Comment:
> I think it's time to clean this up. let's create a map that maps object type to methods that are then called here.

Create a map and map class to method, then call map.get(job.getClass()) to get method to execute.